OverlayPlugin.Core/MemoryProcessors/FFXIVMemory.cs

```diff
@@ -113,23 +120,26 @@ private void CloseProcessHandle()
 
         public bool IsValid()
         {
-            if (process != null && process.HasExited)
+            lock (_mutex)
```

Comment:
You should not put `OnProgressChange event` handlers in mutex (when `process != null && process.HasExited`), which may lead to dead locks. Considering move `OnProgressChange` out of mutex.